### PR TITLE
Ensure errkind_str is not used uninitialized

### PR DESCRIPTION
**Pull request by pmatos**

### CODE REVIEW

File: racket/src/racket/src/error.c [hunk 1]
```diff
@@ -608,6 +609,8 @@ static intptr_t sch_vsprintf(char *s, intptr_t maxlen, const char *msg, va_list
 #endif
 	      tlen = strlen(es) + 24;
 	      t = (const char *)scheme_malloc_atomic(tlen);
+
+              assert(errkind_str);
```
Comment (samth) [inline racket/src/racket/src/error.c]: Did you mean to include this assert?

**Comment by pmatos** [comment]
Didn't I do that?

**Comment by pmatos** [comment]
Oh no... My editor tried to fix all the spacing and when I chose the hunk for the pr, I forgot the hunk with the include. Sorry. Coming soon.

File: racket/src/racket/src/error.c (hunk 1 above)
Comment (pmatos) [reply racket/src/racket/src/error.c]: I did.

Comment (pmatos) [reply racket/src/racket/src/error.c]: See out-of-thread comments.

**Comment by pmatos** [comment]
> Oh no... My editor tried to fix all the spacing and when I chose the hunk for the pr, I forgot the hunk with the include. Sorry. Coming soon.

What am I talking about?!? Now back on the laptop - I can see I did correctly select the right hunks to put into the PR. The `#include <assert.h>` is there.

@samth  ^^^

**Comment by mflatt** [comment]
There's a `MZ_ASSERT` macro that you could use instead of `assert` (and `#include <assert,h>`).

Currently, `MZ_ASSERT` is enabled only on Mac OS, but enabling on Linux would make sense.

**Comment by pmatos** [comment]
I hadn't noticed an `MZ_ASSERT`. Let me have a look at it and get you a new PR.

**Comment by pmatos** [comment]
For this to work lets merge #2852 before.
Waiting for go ahead on your side.

**Comment by pmatos** [comment]
Once this is merged - I will finally enable `-Werror` for CGC CI builds on x86_64.

**Comment by pmatos** [comment]
Thanks - I guess we can actually merge this. It just won't assert anything on Linux... yet - until #2852 is merged.

**Pull request merged**